tests/integration_test/transport_security_test/transport_security_test.go

```diff
@@ -162,73 +163,182 @@ func TestServiceControlWithTLS(t *testing.T) {
 			} else if !strings.Contains(resp, tc.wantResp) {
 				t.Errorf("Test (%s): failed, expected: %s, got: %s", tc.desc, tc.wantResp, resp)
 			}
-		}()
+		})
 	}
 }
 
-func TestHttpsClients(t *testing.T) {
+func TestDownstreamMTLS(t *testing.T) {
 	t.Parallel()
-	args := utils.CommonArgs()
-	args = append(args, fmt.Sprintf("--ssl_server_cert_path=%v", platform.GetFilePath(platform.TestDataFolder)))
-
-	s := env.NewTestEnv(platform.TestHttpsClients, platform.EchoSidecar)
-	defer s.TearDown(t)
-
-	if err := s.Setup(args); err != nil {
-		t.Fatalf("fail to setup test env, %v", err)
-	}
 
 	testData := []struct {
-		desc         string
-		httpsVersion int
-		certPath     string
-		port         uint16
-		wantResp     string
-		wantError    error
+		desc                  string
+		certPath              string
+		getClientCred         func(info *tls.CertificateRequestInfo) (*tls.Certificate, error)
+		httpVersions          []int
+		scheme                string
+		sslServerCertPath     string
+		sslServerRootCertPath string
+		wantError             error
+		wantResp              string
 	}{
 		{
-			desc:         "Succcess for HTTP1 client with TLS",
-			httpsVersion: 1,
-			certPath:     platform.GetFilePath(platform.ServerCert),
-			wantResp:     `simple get message`,
+			desc:                  "failure(client authn),  client doesn't set up cert",
+			scheme:                "http",
+			httpVersions:          []int{1},
+			sslServerRootCertPath: platform.GetFilePath(platform.DownstreamClientCert),
+			getClientCred:         nil,
+			wantError:             fmt.Errorf(`tls: unknown certificate authority`),
 		},
 		{
-			desc:         "Succcess for HTTP2 client with TLS",
-			httpsVersion: 2,
+			desc:                  "failure(client authn),  client cert is unknown to proxy's root cert",
+			scheme:                "http",
+			httpVersions:          []int{1},
+			sslServerRootCertPath: platform.GetFilePath(platform.DownstreamClientCert),
+			getClientCred: func(info *tls.CertificateRequestInfo) (*tls.Certificate, error) {
+				c, err := tls.LoadX509KeyPair(platform.GetFilePath(platform.MismatchCert), platform.GetFilePath(platform.MismatchKey))
```

Comment:
It seems the code for getClientCred are all the same,  just different pairs of client_cert files.  Maybe moving this code into client.DoHttpsGet

Reply:
Done.